### PR TITLE
Implement User Recovery Data deletion on Tenant Deletion

### PR DESCRIPTION
**Pull request by sumedhe**
### Proposed changes in this pull request

Implement the deletion of User Recovery Data in the tenant deletion process.

Fixes: #387

### CODE REVIEW

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/listener/TenantManagementListener.java [hunk 1]
```diff
@@ -95,7 +97,13 @@ public int getListenerOrder() {
     }
 
     @Override
-    public void onPreDelete(int i) throws StratosException {
+    public void onPreDelete(int tenantId) throws StratosException {
 
+        try {
+            UserRecoveryDataStore userRecoveryDataStore = JDBCRecoveryDataStore.getInstance();
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/listener/TenantManagementListener.java:103]: Can there be some other UserRecoveryDataStore which is not type JDBCRecoveryDataStore?

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/store/JDBCRecoveryDataStore.java [hunk 2]
```diff
@@ -305,6 +305,36 @@ public void invalidate(User user) throws IdentityRecoveryException {
         }
     }
 
+    /**
+     * Delete all recovery data by tenant id
+     *
+     * @param tenantId Id of the tenant
+     */
+    public void deleteRecoveryDataByTenantId(int tenantId) throws IdentityRecoveryException {
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/store/JDBCRecoveryDataStore.java:313]: Missing override annotation

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/listener/TenantManagementListener.java (hunk 1 above)
Comment (sumedhe) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/listener/TenantManagementListener.java:103]: JDBCRecoveryDataStore is the only UserRecoveryDataStore implementation used in this component. I think it's fine to use JDBCRecoveryDataStore here. WDYT?

Comment (IsuraD) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/listener/TenantManagementListener.java:103]: Thats is the one we supprt in the latest product by default

**Comment by Kanapriya** [comment]
@sumedhe: Could you please fix these conflicts?

**Pull request merged**